### ISSUE TITLE
In REAPER 7, MIDI inline editor is not focused.

### ISSUE THREAD

**Issue by Aleskis** [jcsteh/osara#933]
Previously, up to version 6.82, MIDI Inline Editor shortcuts were correctly routed to the MIDI Inline Editor. Since version 7, however, this focus has been lost, and keyboard shortcuts are sent back to the TCP, rendering the MIDI Inline Editor unusable.
Of course, there's no vocall feedback, but for quick actions, MIDI Inline Editor works just fine. It would be sad to see a regression in accessibility on what worked in version 6.

**Comment by ScottChesworth** [comment]
I'll take this up with Cockos. Can you give us a list of the in-line actions that you find useful in REAPER 6? While we're sorting this out, it might be possible to get more speech feedback or preview feedback for those.

**Comment by beqabeqa473** [comment]
where it is possible to download reaper 7?

On 9/20/23, ScottChesworth ***@***.***> wrote:
> I'll take this up with Cockos. Can you give us a list of the in-line actions
> that you find useful in REAPER 6? While we're sorting this out, it might be
> possible to get more speech feedback or preview feedback for those.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/jcsteh/osara/issues/933#issuecomment-1726594322
> You are receiving this because you are subscribed to this thread.
>
> Message ID: ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Comment by ScottChesworth** [comment]
Details are at reaper.fm, forum, REAPER Pre-Release Discussion.

**Comment by Aleskis** [comment]
Hey Scott,

I spoke too quickly! Sorry, in fact, the inline midi editor had stopped being focusable, but I hadn't tried switching to full screen, nor had I tried unchecking certain things in the view menu.
On the other hand, if you think we can take vocalization a step further on this subject, I use the actions :

Edit: Note velocity -01
Edit: Note velocity +01
Edit: Note velocity -10
Edit: Note velocity +10
Edit: Move notes down one semitone
Edit: Move notes up one semitone
Edit: Move notes down one octave
Edit: Move notes up one octave
Edit: Move notes left one grid unit
Edit: Move notes right one grid unit
Edit: Move notes left one pixel
Edit: Move notes right one pixel
Edit: Shorten notes one grid unit
Edit: Lengthen notes one grid unit
Edit: Shorten notes one pixel
Edit: Lengthen notes one pixel
Select note nearest to the edit cursor

Note that I use this last action, because there's no equivalent to the actions, move to previous chord and move to next chord in the inline. I don't know what OSARA could implement in this respect, as has been done in the standard editor. By the way, strange that REAPER doesn't provide a native action for moving to the previous or next note via actions. But that's another subject.

**Comment by Aleskis** [comment]
Would you like that I change the issue title ?

**Comment by ScottChesworth** [comment]
> I hadn't tried switching to full screen, nor had I tried unchecking certain things in the view menu.
> 
What needed to be unchecked for the behaviour you're used to to work in REAPER 7?

**Comment by Aleskis** [comment]
In fact, nothing special is checked, except Transport. I suspect that a floating plugin window were masking the inline MIDI editor, because I can't reproduce the focus issue.

**Comment by Aleskis** [comment]
For REAPER 7, I think we could ask Cockos to have the API report what's happening in the MIDI Inline Editor, which would make it easier to work with this tool and, above all, to create an OSARA action that moves from chord to chord, just as you can do in the standard MIDI Editor.

**Comment by RDMurray** [comment]
I don't think there is anything missing from the api for the inline editor, so it should be possible to add feedback however I'm not certain.  Before starting work on it, can you convince me that there is any advantage to the inline editor for screen reader use? As far as I can tell it would take exactly the same number of keystrokes. Also, he inline editor has a much smaller list of actions available.

**Comment by ScottChesworth** [comment]
Have I understood it that in-line would provide us with a more track-based experience for tweaking MIDI? If so, it would certainly be handy here where I tend to do MIDI in short bursts, and consequently find myself opening/closing the editor a lot.

**Comment by ptorpey** [comment]
Is the midi in-line editor something new in version 7?  I didn’t see any documentation for RC 7 on the dev web site.  Does the documentation come with the install?

 

Thanks.

 

--Pete

**Comment by Aleskis** [comment]
Exactly, the main advantage of inline is that you can work on MIDI without having to go into the editor. In this way, you retain access to easy movement between tracks, to the simplified OSARA parameter window, and so on. For quick actions, it's really handy.

**Comment by Aleskis** [comment]
To tell the truth, inline isn't essential, but I use it because I can manage most of my track shortcuts, volume, panning and FX, without having to close the MIDI editor. At the same time, I can also edit my notes as I wish. So I have both mixing and MIDI editing capabilities at my fingertips, just with my keyboard.
However, as I thought, to make it accessible, I'll have to ask Cockos.
